src/frontend/index.ts

```diff
@@ -2,19 +2,38 @@ import sslify from 'express-sslify';
 import helmet from 'helmet';
 import getFrontendApp from './app';
 import {Application} from 'probot';
-import {Express, Router} from 'express';
+import {Express, NextFunction, Request, Response, Router} from 'express';
+import crypto from 'crypto';
 
 function secureHeaders(router: Router, frontendApp: Express) {
+
+  router.use((_: Request, res: Response, next: NextFunction): void => {
+    res.locals.nonce = crypto.randomBytes(16).toString('hex');
+    next();
+  });
+
+
   // Content Security Policy
   router.use(helmet.contentSecurityPolicy({
     directives: {
       defaultSrc: ["'none'"],
+      //Doesn't allow embeddable objects
+      objectSrc: ["'none'"],
       // Allow <script> tags hosted by ourselves and from atlassian when inserted into an iframe
-      scriptSrc: ["'self'", process.env.APP_URL, 'https://*.atlassian.net', 'https://*.jira.com', 'https://connect-cdn.atl-paas.net/'],
+      scriptSrc: ["'self'", process.env.APP_URL, 'https://*.atlassian.net', 'https://*.jira.com', 'https://connect-cdn.atl-paas.net/',
+        "'unsafe-inline'", "'strict-dynamic'", (_: Request, res: Response): string => {
+          return `'nonce-${res.locals.nonce}'`
+        }],
       // Allow XMLHttpRequest/fetch requests
       connectSrc: ["'self'", process.env.APP_URL],
       // Allow <style> tags hosted by ourselves as well as style="" attributes
       styleSrc: ["'self'", "'unsafe-inline'"],
+      // Allow using github-for-jira pages as iframes only in jira
+      frameAncestors: ['https://*.atlassian.net', 'https://*.jira-dev.com'],
```

Comment:
would it ever be needed to also use atlassian.com?

Reply:
Good point. I've checked with the connect team. Currently we only have *.atlassian.net and *.jira.com in prod

Reply:
:+1: